src/Microsoft.Graph/Models/Extensions/DateTimeZoneExtensions.cs

```diff
@@ -0,0 +1,182 @@
+                // Ex: 2019-11-03T01:30:00 in the Pacific time zone occurs twice.  First at 1:30 PDT (-07:00), then an hour later
+                //     at 1:30 PST (-08:00).  We choose PDT (-07:00) because it comes first sequentially.
+                offset = TimeSpan.FromMinutes(Math.Max(offsets[0].TotalMinutes, offsets[1].TotalMinutes));
+            }
+            else if (timeZoneInfo.IsInvalidTime(dateTime))
+            {
+                // Invalid times happen during the gap created by a forward transition, such as the start of daylight saving time.
+                // While they are not values that actually occur in correct local time, they are sometimes encountered
+                // in scenarios such as a scheduled daily task.  In such cases, a reasonable common practice is to advance
+                // to a valid local time by the amount of the transition gap (usually 1 hour).
+                // Ex: 2019-03-10T02:30:00 does not exist in Pacific time.
+                //     The local time went from 01:59:59 PST (-08:00) directly to 03:00:00 PDT (-07:00).
+                //     We will advance by 1 hour to 03:30:00 which is in PDT (-07:00).
+
+                // The gap is usually 1 hour, but not always - so it must be calculated
+                TimeSpan earlierOffset = timeZoneInfo.GetUtcOffset(dateTime.AddDays(-1));
+                TimeSpan laterOffset = timeZoneInfo.GetUtcOffset(dateTime.AddDays(1));
+                TimeSpan gap = laterOffset - earlierOffset;
+
+                dateTime = dateTime.Add(gap);
+                offset = laterOffset;
+            }
+            else
+            {
+                dateTime = DateTime.SpecifyKind(dateTime, DateTimeKind.Utc);
+                offset = timeZoneInfo.GetUtcOffset(dateTime);
+            }
+
+            return new DateTimeOffset(dateTime, offset);
+        }
+    }
+}
+
+namespace Microsoft.Graph
+{
+    public partial class DateTimeTimeZone
+    {      
+        internal const string DateTimeFormat = "yyyy-MM-ddTHH:mm:ss.fffffffK";
+        internal const string DateTimeFormatNonUTC = "yyyy-MM-dd HH:mm:ss tt";
+
+        /// <summary>
+        /// Converts a Datetime parameter to its equivalent DateTimeTimeZone Complex Type given its TimeZone
+        /// </summary>
+        /// <param name="dateTime"></param>
+        /// <param name="timeZone"></param>
+        /// <returns></returns>
+        public static DateTimeTimeZone FromDateTime(DateTime dateTime, string timeZone)
+        {
+            TimeZoneInfo timeZoneInfo = GetTimeZoneInfo(timeZone);
+
+            // When in Local or Utc kind, we can convert to the time zone specified
+            // Which may be a no-op if the value is already in the specified time zone
+            // We don't do this for Unspecified kind, as that would infer converting *from* the local time zone
+            if (dateTime.Kind != DateTimeKind.Unspecified)
```

Comment:
We don't want to do any checking. We want to just help create a DateTimeTimeZone

Reply:
Delete lines 119 through 127. We don't want to get in the time zone business since we would have to master DateTime and time zones across dozens of workloads, and considering that we support Xamarin, across Windows, iOS, Android, and Linux, and Mac.